R/install-git.R

```diff
@@ -148,8 +148,42 @@ remote_package_name.git2r_remote <- function(remote, ...) {
   if (grepl("^https?://", remote$url)) {
     # assumes GitHub-style "<repo>/raw/<ref>/<path>" url
     url <- build_url(sub("\\.git$", "", remote$url), "raw", remote_sha(remote, ...), description_path)
+    download_args <- list(path = tmp, url = url)
+    if (!is.null(remote$credentials)) {
+      if (inherits(remote$credentials, "cred_user_pass")) {
+        download_args$basic_auth <- list(
+          user = remote$credentials$username,
+          password = remote$credentials$password
+        )
+      } else if (inherits(remote$credentials, "cred_env")) {
+        if (Sys.getenv(remote$credentials$username) == "") {
+          stop(paste0("Environment variable `", remote$credentials$username, "` is unset."), .call = FALSE)
+        }
+        if (Sys.getenv(remote$credentials$password) == "") {
+          stop(paste0("Environment variable `", remote$credentials$password, "` is unset."), .call = FALSE)
+        }
+        download_args$basic_auth <- list(
+          user = Sys.getenv(remote$credentials$username),
+          password = Sys.getenv(remote$credentials$username)
```

Comment:
There is a missing parenthesis there
```suggestion
          password = Sys.getenv(remote$credentials$username)
       )
```